internal/util/update/update.go

```diff
@@ -176,6 +186,23 @@ func (u Command) Run(ctx context.Context) error {
 	return nil
 }
 
+// updateSubKf updates subpackage with given ref and update strategy
+func updateSubKf(subKf *kptfilev1.KptFile, ref string, strategy kptfilev1.UpdateStrategyType) {
+	// check if explicit ref provided
+	if ref != "" {
+		subKf.Upstream.Git.Ref = ref
+	}
+	if strategy != "" {
+		subKf.Upstream.UpdateStrategy = strategy
+	}
+}
+
+// isPkgInRootPkg checks if a pkg is a subpkg of root pkg or is root pkg.
+// This is true if pkg has the same upstream repo and upstream directory is within or equal to root pkg directory
+func isPkgInRootPkg(subKf, rootKf *kptfilev1.KptFile) bool {
+	return subKf.Upstream.Git.Repo == rootKf.Upstream.Git.Repo && strings.HasPrefix(subKf.Upstream.Git.Directory, rootKf.Upstream.Git.Directory)
```

Comment:
Is HasPrefix reliable ? what if `subKf.Upstream.Git.Directory = foo` and `rootKf.Upstream.Git.Directory = foo/`

Reply:
Do we expect users to edit `Upstream.Git.Directory`? I did not add an additional clean because `get` does it. 
https://github.com/GoogleContainerTools/kpt/blob/faf69b23ac6df9740e2c0ee7232ca70f68b70b5f/internal/util/parse/parse.go#L80

Reply:
Users can handcraft/edit upstream section. So better to clean the path for comparison.

Reply:
got it, added